### PR TITLE
Solicitor create case

### PR DESCRIPTION
**Pull request by MOsinloye**

### CODE REVIEW

File: pages/SolAwaitingPaymentConfirmationPage.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const I = actor();
+
+module.exports = {
+
+  fields: {
+    signOut: '#sign-out',
+  },
+
+  fillFormAndSubmit() {
```
Comment (hosainnet) [inline pages/SolAwaitingPaymentConfirmationPage.js]: Should this be `signOut`?

File: tests/solicitorCreateCase.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    console.log(caseNumber);
+    I.statementOfTruthAndReconciliationPageFormAndSubmit();
+    I.casePaymentAndSubmissionPageFormAndSubmit();
+    I.caseOrderSummaryPageFormAndSubmit();
+    I.caseApplicationCompletePageFormAndSubmit();
+    I.caseCheckYourAnswersPageFormAndSubmit();
+    I.solAwaitingPaymentConfPageFormAndSubmit();
+    I.wait(4);
+    // I.click('Sign Out');
+    // I.login(caseWorker.username, caseWorker.password);
```
Comment (hosainnet) [inline tests/solicitorCreateCase.test.js]: Should we remove this commented out code?

File: tests/solicitorCreateCase.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    I.selectReasonForTheDivorceQuestionPageAndSubmit();
+    I.fillAdulteryDetailsFormAndSubmit();
+    I.fillAdulteryDetailsSecondPageFormAndSubmit();
+    I.otherLegalProceedings();
+    I.financialOrdersSelectButton();
+    I.claimForCostsSelectButton(),
+    I.uploadTheMarriageCertificateOptional();
+    I.solicitorCreateCheckYourAnswerAndSubmit();
+    const caseNumber = await I.solicitorCaseCreatedAndSubmit();
+    console.log(caseNumber);
```
Comment (hosainnet) [inline tests/solicitorCreateCase.test.js]: Is this needed anymore?

File: pages/AboutSolicitorPage.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  fillFormAndSubmit() {
+    I.fillField(this.fields.solicitorsName, 'PAPA AJASCO');
+    I.fillField(this.fields.firmName, 'ELEMU GETON');
+    I.fillField(this.fields.firmDxAddress, '100 Reede Road, RM10 8DU');
+    I.fillField(this.fields.yourReferenceNumber, 'AWS11234');
+    I.fillField(this.fields.phoneNumber, '07712345679');
+    I.fillField(this.fields.email, 'ccdsolicitorcase@pettyfrance.com');
+    I.click(this.fields.solicitorAgree);
+    I.wait(5)
```
Comment (hosainnet) [inline pages/AboutSolicitorPage.js]: We have quite a lot of `wait` statements. Is `waitForAction` in codeceptjs config not enough?

File: tests/solicitorCreateCase.test.js (hunk 3 above)
Comment (MOsinloye) [reply tests/solicitorCreateCase.test.js]: Yes they are, I need the casenumber for my next story

File: pages/AboutSolicitorPage.js (hunk 4 above)
Comment (MOsinloye) [reply pages/AboutSolicitorPage.js]: waitForAction is not enough. But I'll look into the wait() and try to reduce them for subsequents tests

File: tests/solicitorCreateCase.test.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+const { getSolicitorLoginDetails, getCaseWorkerLoginDetails } = require('../helpers/utils');
+
+
+const solicitor = getSolicitorLoginDetails();
+const caseWorker = getCaseWorkerLoginDetails();
+
+Feature('Solicitor create case');
+
+Scenario('Solicitor create case and case worker views case', async (I) => {
```
Comment (vianvi) [inline tests/solicitorCreateCase.test.js]: Is caseworker viewing case

File: helpers/utils.js [hunk 6]
```diff
@@ -1,18 +1,29 @@
-const getTestUserLoginDetails = () => {
-    if (!process.env.CCD_E2E_PASSWORD) {
-        throw new Error('You need to set CCD_E2E_PASSWORD env variable');
+const getSolicitorLoginDetails = () => {
+    if (!process.env.CCD_E2E_EMAIL || !process.env.CCD_E2E_PASSWORD) {
```
Comment (alexwakeman) [inline helpers/utils.js]: So in the other `div` projects we are using the infrastructure files to inject environment variables, and applying them to a `conifg.yml` file to be accessed in application code. You can see an example of how the infrastructure files pull in `var`s from the Azure vault [here](https://github.com/hmcts/div-respondent-frontend/blob/master/infrastructure/main.tf):
```
IDAM_API_URL = "${var.idam_api_url}"
```
And then in the `config/default.yml`:
```
services:
    idam:
        apiUrl: IDAM_API_URL
```

Then we can use the npm module called `[config](https://github.com/lorenwest/node-config/wiki/Configuration-Files)` to read the yaml and use it application code. 

Doing all this means we don't need to maintain environment variables in the various pipelines this application may run in. We can also utilise the Azure vault for secrets and so on.

Comment (MOsinloye) [reply helpers/utils.js]: Thanks @alexwakeman , as discussed I'll need your help to do this in the next story. Appreciate

**Pull request merged**